File: src/Entity/CourseVideo.php
<?php

namespace App\Entity;

use App\Repository\CourseVideoRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: CourseVideoRepository::class)]
class CourseVideo
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    private ?string $title = null;

    #[ORM\Column(length: 255)]
    private ?string $videoUrl = null;

    #[ORM\Column]
    private ?int $durationSec = null;

    #[ORM\Column]
    private ?int $orderIndex = null;

    #[ORM\Column]
    private ?bool $isPreview = null;

    #[ORM\ManyToOne(inversedBy: 'courseVideos')]
    #[ORM\JoinColumn(nullable: false)]
    private ?CourseSection $section = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTitle(): ?string
    {
        return $this->title;
    }

    public function setTitle(string $title): static
    {
        $this->title = $title;

        return $this;
    }

    public function getVideoUrl(): ?string
    {
        return $this->videoUrl;
    }

    public function setVideoUrl(string $videoUrl): static
    {
        $this->videoUrl = $videoUrl;

        return $this;
    }

    public function getDurationSec(): ?int
    {
        return $this->durationSec;
    }

    public function setDurationSec(int $durationSec): static
    {
        $this->durationSec = $durationSec;

        return $this;
    }

    public function getOrderIndex(): ?int
    {
        return $this->orderIndex;
    }

    public function setOrderIndex(int $orderIndex): static
    {
        $this->orderIndex = $orderIndex;

        return $this;
    }

    public function isPreview(): ?bool
    {
        return $this->isPreview;
    }

    public function setIsPreview(bool $isPreview): static
    {
        $this->isPreview = $isPreview;

        return $this;
    }

    public function getSection(): ?CourseSection
    {
        return $this->section;
    }

    public function setSection(?CourseSection $section): static
    {
        $this->section = $section;

        return $this;
    }
}

File: src/main/java/com/pegasus/entities/CourseVideo.java
package com.pegasus.entities;

public class CourseVideo {
    private int id;
    private String title;
    private String videoUrl;
    private int durationSec;
    private int orderIndex;
    private boolean preview;
    private int sectionId;

    public CourseVideo() {
    }

    public CourseVideo(int id, String title, String videoUrl, int durationSec, int orderIndex, boolean preview, int sectionId) {
        this.id = id;
        this.title = title;
        this.videoUrl = videoUrl;
        this.durationSec = durationSec;
        this.orderIndex = orderIndex;
        this.preview = preview;
        this.sectionId = sectionId;
    }

    public CourseVideo(String title, String videoUrl, int durationSec, int orderIndex, boolean preview, int sectionId) {
        this.title = title;
        this.videoUrl = videoUrl;
        this.durationSec = durationSec;
        this.orderIndex = orderIndex;
        this.preview = preview;
        this.sectionId = sectionId;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getVideoUrl() {
        return videoUrl;
    }

    public void setVideoUrl(String videoUrl) {
        this.videoUrl = videoUrl;
    }

    public int getDurationSec() {
        return durationSec;
    }

    public void setDurationSec(int durationSec) {
        this.durationSec = durationSec;
    }

    public int getOrderIndex() {
        return orderIndex;
    }

    public void setOrderIndex(int orderIndex) {
        this.orderIndex = orderIndex;
    }

    public boolean isPreview() {
        return preview;
    }

    public void setPreview(boolean preview) {
        this.preview = preview;
    }

    public int getSectionId() {
        return sectionId;
    }

    public void setSectionId(int sectionId) {
        this.sectionId = sectionId;
    }

    @Override
    public String toString() {
        return "CourseVideo{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", durationSec=" + durationSec +
                ", orderIndex=" + orderIndex +
                ", preview=" + preview +
                ", sectionId=" + sectionId +
                '}';
    }
}
